### ISSUE TITLE
Look at differing between dynamic and static locations and how we can update the location on a regular cadence from APIs

### ISSUE THREAD

**Issue by einari** [dolittle-platform/edge#21]
Some locations are static - typically buildings, while others are dynamic / moving. An example of this is ships. 

In an ideal world the location that is moving should report back to system where it is. Unfortunately this is not possible to know for everything and even those locations that has an exact GPS position on-board, might not have connectivity all the time,

It would be therefor be great to have a way of definining different types of APIs that can be called with a regular cadence to get the exact location.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1115616103797671/1115621132467096)

**Comment by jakhog** [comment]
I think we should think a bit about location vs. position, for a few reasons:
1. Position is sometimes tricky to represent, the WGS84 coordinates (from GPS) is not always perfect, e.g. in buildings where floor, room makes more sense.
2. We should definitely allow edge nodes to _not_ have a recorded position, for privacy reasons.
3. It probably makes sense to group nodes across different positions, and probably in some kind of hierarchy.

IMHO, we should at least not make (lat,lon) a feature of all edge nodes.

**Comment by einari** [comment]
Good point. The differentation of static nodes and dynamic would to a certain degree capture some of this. A static node would be a building and not report geo location - you set it up, while a dynamic could report position. But as you say, it shouldn't be required to have position.
